libvirt/tests/src/passthrough/pci/libvirt_pci_passthrough.py

```
@@ -207,19 +207,83 @@ def check_device_status(net_ip, server_ip, netmask):
+                                    nic_name = iface
+                                    logging.info("Found interface %s for PCI %s using uevent method", nic_name, val)
+                                    break
+                    except Exception as e:
+                        logging.error("Failed to find interface for PCI %s: %s", val, str(e))
+
+                if nic_name == "None" or not nic_name:
+                    test.error("Could not determine interface name for PCI device {}".format(val))
+                    continue
+
+                # Convert netmask to CIDR
+                cidr_mask = netmask_to_cidr(netmask)
```

Ruff (0.15.20)

[error] Undefined name `netmask_to_cidr`

(F821)